Realm/RLMRealm.mm

```diff
@@ -1043,4 +1045,29 @@ - (void)detachAllEnumerators {
     _collectionEnumerators = nil;
 }
 
+- (RLMSyncSubscriptionSet *)subscriptions {
+    return [self subscriptionsWithError:nil];
+}
+
+- (RLMSyncSubscriptionSet *)subscriptionsWithError:(NSError **)error {
+    if (_realm->config().sync_config) {
+        if (_realm->config().sync_config->flx_sync_requested) {
+#if REALM_ENABLE_SYNC
+            return [[RLMSyncSubscriptionSet alloc] initWithSubscriptionSet:_realm->get_latest_subscription_set() realm:self];
+#else
+            NSLog(@"Realm was not built with sync enabled");
```

Comment:
If this is erroneous then it should throw an exception, and if it's fine then it shouldn't log anything.